app/api/event_invoices.py

```diff
@@ -0,0 +1,110 @@
+
+    id = fields.Str(dump_only=True)
+    identifier = fields.Str()
+    amount = fields.Float()
+    address = fields.Str()
+    city = fields.Str()
+    state = fields.Str()
+    country = fields.Str()
+    zipcode = fields.Str()
+    created_at = fields.DateTime()
+    completed_at = fields.DateTime(default=None)
+    transaction_id = fields.Str()
+    paid_via = fields.Str()
+    payment_mode = fields.Str()
+    brand = fields.Str()
+    exp_month = fields.Integer()
+    exp_year = fields.Integer()
+    last4 = fields.Str()
+    stripe_token = fields.Str()
+    paypal_token = fields.Str()
+    status = fields.Str()
+    user = Relationship(attribute='user',
+                        self_view='v1.event_invoice_user',
+                        self_view_kwargs={'id': '<id>'},
+                        related_view='v1.user_detail',
+                        related_view_kwargs={'event_invoice_id': '<id>'},
+                        schema='UserSchema',
+                        type_='user')
+    event = Relationship(attribute='event',
+                         self_view='v1.event_invoice_event',
+                         self_view_kwargs={'id': '<id>'},
+                         related_view='v1.event_detail',
+                         related_view_kwargs={'event_invoice_id': '<id>'},
+                         schema='EventSchema',
+                         type_='event')
+    discount_code = Relationship(attribute='discount_code',
+                         self_view='v1.event_invoice_discount_code',
+                         self_view_kwargs={'id': '<id>'},
+                         related_view='v1.discount_code_detail',
+                         related_view_kwargs={'event_invoice_id': '<id>'},
+                         schema='DiscountCodeSchema',
+                         type_='discount_code')
+
+
+class EventInvoiceDetail(ResourceDetail):
+    """
+    Event Invoice detail by id
+    """
+    decorators = (jwt_required, )
+    schema = EventInvoiceSchema
+    data_layer = {'session': db.session,
+                  'model': EventInvoice}
+
+
+class EventInvoiceList(ResourceList):
+    """
+    List and Create Event Invoices
+    """
+
+    def query(self, view_kwargs):
+        query_ = self.session.query(EventInvoice)
+        if view_kwargs.get('event_id') is not None:
+            query_ = query_.filter_by(event_id=view_kwargs['event_id'])
+        if view_kwargs.get('user_id') is not None:
+            query_ = query_.filter_by(user_id=view_kwargs['user_id'])
+        return query_
+
+    def before_create_object(self, data, view_kwargs):
+        if view_kwargs.get('event_id') is not None:
+            event = self.session.query(Event).filter_by(id=view_kwargs['event_id']).one()
+            data['event_id'] = event.id
+        if view_kwargs.get('user_id') is not None:
+            user = self.session.query(User).filter_by(id=view_kwargs['user_id']).one()
+            data['user_id'] = user.id
+
+    decorators = (jwt_required, )
```

Comment:
`is_super_admin` is okay for now, but in future, a two-factor authentication endpoint should be setup so that the super admin does not accidentally edit any info by mistake via the api